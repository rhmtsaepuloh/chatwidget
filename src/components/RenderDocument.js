import React from "react";
import { MdFileDownload, MdDoneAll } from "react-icons/md";
import { formatBytes, capitalize } from "../utils/memory";

const RenderDocument = ({
  icon,
  text,
  extension,
  splitExtension,
  size,
  timestamp,
  downloadFile,
  caption,
}) => {
  return (
    <React.Fragment>
      <div className='file-document flex items-center justify-between bg-blue-200 w-full rounded px-1 py-3'>
        <div className='flex items-center  w-3/4'>
          {icon}
          <p className='whitespace-pre-wrap break-words ml-2 w-3/4 text-xs'>
            {text}
          </p>
        </div>
        <MdFileDownload
          onClick={() => downloadFile(extension, text)}
          className='text-blue-700 text-xl cursor-pointer'
        />
      </div>
      <p className='whitespace-pre-wrap break-words mb-3'>{caption}</p>
      <div className='file-timestamp flex items-center justify-between'>
        <p className='text-sm' style={{ color: "#979797" }}>
          {capitalize(splitExtension)}
          <label className='mx-1'>&bull;</label>
          {formatBytes(size)}
        </p>
        <div className='flex justify-end'>
          <p className='text-sm' style={{ color: "#979797" }}>
            {timestamp}
          </p>
          <MdDoneAll className='ml-1' />
        </div>
      </div>
    </React.Fragment>
  );
};

export default RenderDocument;
