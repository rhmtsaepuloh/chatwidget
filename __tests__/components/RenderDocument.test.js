/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";
import RenderDocument from "../../src/components/RenderDocument";
import { FaFilePdf } from "react-icons/fa";

const props = {
  icon: <FaFilePdf />,
  text: "text",
  extension: "pdf",
  splitExtension: "file.pdf",
  size: 180,
  timestamp: "10:20",
  downloadFile: jest.fn(),
  caption: "test",
};

describe("render correctly render document components", () => {
  test("renders", () => {
    const wrapper = shallow(<RenderDocument {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
