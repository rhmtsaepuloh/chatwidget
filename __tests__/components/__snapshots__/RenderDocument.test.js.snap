// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly render document components renders 1`] = `ShallowWrapper {}`;
